### PR TITLE
Pr automation

### LINKED ISSUE #1069: Automate some of the PR things
@Relequestual requested that some things be automated.  One of those things included ensuring that an issue be associated with PRs.  This is the issue associated with the PR that adds automation to ensure that issues are associated to PRs.

### PR DESCRIPTION
**Pull request by gregsdennis**
Resolves #1069 
Relates to #1070 & #1071

Happy to take opinions on the content of 👇.  It all works pretty well, though, so I don't think I'll be changing the mechanics.

---

- [x] [Issue](https://github.com/json-schema-org/json-schema-spec/issues) linked
- [x] [Milestone](https://github.com/json-schema-org/json-schema-spec/milestones) assigned
- [x] Test coverage issue in [test suite](https://github.com/json-schema-org/JSON-Schema-Test-Suite)

### CODE REVIEW

**Comment by gregsdennis** [comment]
adding checklist look 👍 

working on enforcing next...

**Comment by karenetheridge** [comment]
can you summarize what these checklists entail, for those of us who haven't learned github workflows yet? :)

**Comment by gregsdennis** [comment]
There's some discussion in #1069 around it.  It's basically a guide or reminder.

- Not having all the items checked will fail the build, but it won't prevent merging.
- The build doesn't auto-run when the opening comment changes, so it'll need to be re-run manually.
- It's not really enforcement since items can be removed or falsely checked.

**Comment by jdesrosiers** [comment]
Honestly, I think this work is a bit premature. Our process should be discussed, agreed upon, and documented in `CONTRIBUTING.md` before we automate it. Having an assigned milestone and a test suite issue are requirements (recommendations?) that are not currently documented. Linking an issue is only mentioned as something that should be done _if_ there is an issue that the PR resolves. Let's start by documenting what we want, then add automation where it makes sense.

**Comment by gregsdennis** [comment]
As I mentioned above, there has been some discussion in issues as well as in Slack, but if you'd like official docs first that's fine. I don't think this will go stale or conflict with anything.

**Comment by Relequestual** [comment]
> Honestly, I think this work is a bit premature. Our process should be discussed, agreed upon, and documented in `CONTRIBUTING.md` before we automate it. Having an assigned milestone and a test suite issue are requirements (recommendations?) that are not currently documented. Linking an issue is only mentioned as something that should be done _if_ there is an issue that the PR resolves. Let's start by documenting what we want, then add automation where it makes sense. - @jdesrosiers 

I think I'd have to agree. I'm doing some work this week on community things so let's see how we go.

**Review by 1050880 — COMMENTED** [review]
@afshin4030

**Comment by handrews** [comment]
@gregsdennis given the concerns from @jdesrosiers and @Relequestual (with which I think I agree), can this be closed in favor of a discussion somewhere?

**Comment by gregsdennis** [comment]
There's already discussion elsewhere, as noted in the opening comment.

This PR isn't doing any harm just sitting here.  This kind of automation is still something I'd like to see.  It relates to repo management, like requiring an approval, which we did a few weeks ago; it's a nice to have.  The only reason it's still open is that people have forgotten about it.

**Comment by gregsdennis** [comment]
I also discovered that [actions can trigger from comment changes](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issue_comment), so I might add that.

**Comment by gregsdennis** [comment]
Closing per https://github.com/json-schema-org/json-schema-spec/issues/1069#issuecomment-2175677480
